lib/lib-table-query/src/main/java/org/sagebionetworks/table/query/model/ReplaceableBox.java

```diff
@@ -0,0 +1,59 @@
+package org.sagebionetworks.table.query.model;
+
+import java.util.Objects;
+
+/**
+ * Box of an Element that be be replaced. *
```

Comment:
Let us extend the docs to explain why we need this boxing in some situation (e.g. when the parent element might different sub-components of different types)